components/gitpod-protocol/src/billing-mode.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+/**
+ * BillingMode is used to answer the following questions:
+ *  - Should UI piece x be displayed for this user/team? (getBillingModeForUser/Team)
+ *  - What model should be used to limit this workspace's capabilities (mayStartWorkspace, setTimeout, workspace class, etc...) (getBillingMode(workspaceInstance.attributionId))
+ *  - How is a workspace session charged for? (getBillingMode(workspaceInstance.attributionId))
+ */
+export type BillingMode = None | Chargebee | UsageBased;
+export namespace BillingMode {
+    export const NONE: None = {
+        mode: "none",
+    };
+
+    export function showChargebeeBilling(billingMode: BillingMode): boolean {
```

Comment:
These functions are not used nor tested, yet, but helped me greatly when thinking about how we want to used BillingMode later on. Left them in for reference, and maybe they're actually helpful